packages/react-composites/tests/browser/call/hermetic/RoomsCallScreen.test.ts

```diff
@@ -63,3 +71,89 @@ test.describe('Rooms CallScreen tests for different roles', async () => {
     expect(await stableScreenshot(page)).toMatchSnapshot('rooms-call-screen-consumer-remote-participants.png');
   });
 });
+
+test.describe('Rooms Participant RemoveButton tests for different roles', async () => {
+  test('Remove button is enabled for Presenter', async ({ page, serverUrl }, testInfo) => {
+    test.skip(isTestProfileStableFlavor());
+    const paul = defaultMockRemoteParticipant('Paul Bridges');
+    const participants = [paul];
+    const initialState = defaultMockCallAdapterState(participants);
+    await page.goto(buildUrlWithMockAdapter(serverUrl, initialState, { role: 'Presenter' }));
+    await waitForSelector(page, dataUiId(IDS.videoGallery));
+
+    if (isTestProfileDesktop(testInfo)) {
+      await waitForSelector(page, dataUiId('call-composite-participants-button'));
+      await pageClick(page, dataUiId('call-composite-participants-button'));
+      await waitForSelector(page, dataUiId('participant-item-menu-button'));
+      await pageClick(page, dataUiId('participant-item-menu-button'));
+    } else {
+      await pageClick(page, dataUiId('call-with-chat-composite-more-button'));
+      await waitForSelector(page, dataUiId('call-composite-more-menu-people-button'));
+      await pageClick(page, dataUiId('call-composite-more-menu-people-button'));
+      await hidePiPiP(page);
+      await waitForSelector(page, dataUiId('participant-list'));
+      await waitForSelector(page, dataUiId('participant-item'));
+      await pageClick(page, dataUiId('participant-item'));
+      await waitForSelector(page, dataUiId('drawer-menu'));
+    }
+    expect(await stableScreenshot(page, { dismissTooltips: true })).toMatchSnapshot(
+      'rooms-call-remove-participant-presenter.png'
+    );
+  });
+
+  test('Remove button is disabled for Attendee', async ({ page, serverUrl }, testInfo) => {
+    test.skip(isTestProfileStableFlavor());
+    const paul = defaultMockRemoteParticipant('Paul Bridges');
+    const participants = [paul];
```

Comment:
All of these tests are the *exact same code* after setting a different role.

Can you extract a helper function that contains all the test logic and call it from all three tests to avoid duplication?

It can just be a free function in this file.

Reply:
Something like `openRemoveParticipantMenu(page)`